### DISCUSSION TITLE
git branch management

### DISCUSSION THREAD

**Discussion by connorferster** [JWock82/Pynite#201] [Ideas]
In reference to the v1.0 issue, I am proposing a new and simple branch management technique that allows releases to be published and patched while also allowing for new features to be developed. I have used this release style myself and have found it incredibly simple and convenient.

Here is a diagram of it:
![image](https://github.com/JWock82/PyNite/assets/38441489/bbc86623-6218-443c-8a34-4a25590f0dbd)

In essence:

* Feature branches are created off of main and features are developed before being merged back into main (current practice on PyNite, as shown in orange dashed rectangle)
* NEW: When a feature is complete (merged back into main), then a new branch is created for the release. The activity that happens in this release branch is as follows:
    * Version numbers are bumped
    * Changelogs are updated (if applicable)
* NEW: The release branch is then merged back into main (this is how main maintains the most up-to-date version) and, importantly, the release branch is NOT deleted (it is **kept** in the repo). A new release is created on GitHub to trigger publishing to PyPI, etc.
* NEW: If an issue comes up with the release and that particular release needs fixing, then a branch is created to fix the bug (off of the release branch), the hotfix is made and tested (one or more commits) and then merged back into main. Again, the release branch is **kept**.
* NEW: A new release is created based on the new release branch to trigger publishing to PyPI, etc.

**Comment by JWock82** [comment]
I like it. I will create a release branch
